src/cls/rgw/cls_rgw.cc

```diff
@@ -523,7 +523,8 @@ static int check_index(cls_method_context_t hctx, struct rgw_bucket_dir_header *
         CLS_LOG(1, "ERROR: rgw_bucket_list(): failed to decode entry, key=%s\n", kiter->first.c_str());
         return -EIO;
       }
-      struct rgw_bucket_category_stats& stats = calc_header->stats[entry.meta.category];
+      struct rgw_bucket_category_stats& stats =
```

Comment:
why the C-style struct notation here?

Reply:
Don't know! Cleaned it up across the check_index function. Looks like it's wide-spread in this file. Maybe another PR can clean all that up.